azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -70,6 +72,17 @@ def decrypt_p7m(self, p7m_file, trans_prv_file, trans_cert_file, pem_file):
         rc = shellutil.run(cmd)
         if rc != 0:
             logger.error("Failed to decrypt {0}".format(p7m_file))
+    
+    def decrypt_encrypted_file(self, privateKey, encryptedFile):
+        cmd = "{0} cms -decrypt -inform DER -inkey {1} -in {2}".format(self.openssl_cmd, privateKey, encryptedFile)
+        output = shellutil.run_get_output(cmd)
+        if output[0] != 0:
+            msg = "Error decrypting file {0}".format(output[1])
+            logger.error(msg)
+            raise CryptError(msg)
+        #return output[1].decode('utf8').replace("\0", "")
+        #TODO: find a better way to clean up the response.
+        return output[1].replace("\0", "").strip()
```

Comment:
You will need to handle this before we commit.  The output should already be sanitized.  What more are you trying to accomplish?

We get bit frequently by Python2 vs. Python3 issues.

Reply:
done